docs/source/app-dev/app-arch.rst

```diff
@@ -293,3 +293,25 @@ As a last resort for these cases, your tests can use the reset service, which le
 The reset service has a single ``reset`` method that will cause all the accumulated state to be dropped, including all active contracts, the entire history of transactions and all allocated users. Only the DAML packages loaded in the ledger are preserved, thereby saving the time needed for reloading them as opposed to simply spinning up a new ledger.
 
 The reset service momentarily shuts down the gRPC channel it communicates over, so your testing infrastructure must take this into account and, when the ``reset`` is invoked, must ensure that tests are temporarily suspended as attempts to reconnect with the rebooted ledger are performed. There is no guarantee as to how long the reset will take, so this should also be taken into account when attempting to reconnect.
+
+.. _handling-submission-failures:
+
+Handle failures when submitting commands
+****************************************
+
+The interaction of a DAML application with the ledger is inherently asynchronous: applications send commands to the ledger, and some time later they see the effect of that command on the ledger.
+
+There are several things that can fail during this time window: the application can crash, the participant node can crash, messages can be lost on the network, or the ledger may be just slow to respond due to a high load. 
+
+If you want to make sure no command is executed twice (especially important with non-consuming choices), your application needs to robustly handle all the various failure scenarios.
+DAML ledgers provide **command deduplication** to solve this problem:
+Applications can assign a “time to live” (TTL) to each command,
+and the ledger will guarantee that duplicate commands (using the same submitter and command ID) will be ignored within this time window.
+
+To use command deduplication, you should:
+
+- Use generous values for the TTL. It should be large enough such that you can assume the command was permanently lost if the TTL has passed and you still don’t observe any effect of the command on the ledger.
```

Comment:
The explanation here makes IMHO less sense. The TTL should be chosen large enough such that the user is sure that the application will not try to resend the command ever again. 
The purpose of the command-id deduplication is that you don't have to explicitly track commands but can just re-submit them without worrying about a previous one.

Reply:
The application will resend commands if:

1. The application crashes and recovers, and does not know whether a given command was already sent.
2. There is no observable effect (no completion event, no change in the ACS) after a long time. The application will have to assume the command was lost by the ledger and retry.


To handle the second case, you need to have a finite duration of the deduplication time window.